frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs

```diff
@@ -171,18 +182,56 @@ createAccountSplash model name chain mPublicKey keysetPresets = Workflow $ do
         pure $ ( constDyn $ mkAccountGuard (Set.singleton key) "keys-all"
                , emptyKeysetPresets
                )
+
+    dialogSectionHeading mempty "Transaction Gas Payer"
+    (_onCreateSelect, dCreateSelect) <- divClass "group" $ mdo
+      dCreate <- holdDyn CreateAccountMethod_SelfGasPayer $ onCreateSelect
+      let
+        mkLbl lbl cls =
+          fst <$> elClass' "span" (renderClass cls) (text lbl)
+
+        mkRadioOption lbl opt = divClass "create-account__gas-payer" $
+          uiLabeledRadioView (mkLbl lbl) dCreate opt
+
+        onCreateSelect =
+          leftmost [onOwn, onExternal, onShare]
+
+      onOwn <- mkRadioOption "My own Chainweaver account (basic transaction)"
+        CreateAccountMethod_SelfGasPayer
+      onExternal <- mkRadioOption "My own external account (requires private key)"
+        CreateAccountMethod_ExternalAccount
+      onShare <- mkRadioOption "Another user’s account (share account creation details)"
+        CreateAccountMethod_ShareTxBuilder
+
+      pure ( onCreateSelect
+           , dCreate
+           )
+
+    pure (ks, ksSelections, dCreateSelect)
+
+  let nextDisabled = ffor2 keyset dCreationMethod $ \mkeys -> \case
+        CreateAccountMethod_ExternalAccount -> False
+        _ -> isNothing mkeys
+
   (cancel, next) <- modalFooter $ do
     cancel <- cancelButton def "Cancel"
-    let cfg = def & uiButtonCfg_disabled .~ fmap isNothing keyset
-    notGasPayer <- confirmButton cfg "I am not the Gas Payer"
-    gasPayer <- confirmButton cfg "I am the Gas Payer"
-    let next = leftmost
-          [ tagMaybe (fmap (createAccountNotGasPayer model name chain mPublicKey keysetSelections)
-                      <$> current keyset) notGasPayer
-          , tagMaybe (fmap (createAccountConfig model name chain mPublicKey keysetSelections)
-                      <$> current keyset) gasPayer
-          ]
-    pure (cancel, next)
+
+    next <- confirmButton (def & uiButtonCfg_disabled .~ nextDisabled) "Next"
+
+    let
+      notGasPayerWF = createAccountNotGasPayer model name chain mPublicKey keysetSelections
+      gasPayerWF = createAccountConfig model name chain mPublicKey keysetSelections
+      externalPayerWF = createAccountFromExternalAccount  model name chain mPublicKey keysetSelections
+
+      nextWF = ffor2 keyset dCreationMethod $ \mkeys -> \case
```

Comment:
`\mkeys -> ffor mkeys $ \case`

Reply:
Why?

Reply:
Ah indeed, the change you did is much better